Doppler.Sap/Factory/SapTaskHandler.cs

```diff
@@ -3,74 +3,93 @@
 using System.Net.Http;
 using System.Text;
 using System.Threading.Tasks;
+using Doppler.Sap.Mappers;
 using Doppler.Sap.Models;
 using Microsoft.Extensions.Logging;
 using Newtonsoft.Json;
 using Microsoft.Extensions.Options;
 
 namespace Doppler.Sap.Factory
 {
-    public abstract class SapTaskHandler
+    public class SapTaskHandler : ISapTaskHandler
     {
-        protected readonly SapConfig SapConfig;
-        protected SapLoginCookies SapCookies;
-        private DateTime _sessionStartedAt;
-        public HttpClient Client;
+        private readonly SapConfig _sapConfig;
+        private SapLoginCookies _sapCookies;
+        private readonly IHttpClientFactory _httpClientFactory;
         private readonly ILogger<SapTaskHandler> _logger;
 
-        protected SapTaskHandler(IOptions<SapConfig> sapConfig, ILogger<SapTaskHandler> logger)
+        public SapTaskHandler(IOptions<SapConfig> sapConfig, ILogger<SapTaskHandler> logger, IHttpClientFactory httpClientFactory)
         {
-            SapConfig = sapConfig.Value;
+            _sapConfig = sapConfig.Value;
             _logger = logger;
-
-            var handler = new HttpClientHandler
-            {
-                ServerCertificateCustomValidationCallback = HttpClientHandler.DangerousAcceptAnyServerCertificateValidator,
-                UseCookies = false
-            };
-
-            Client = new HttpClient(handler);
+            _httpClientFactory = httpClientFactory;
         }
 
-        public abstract Task<SapTaskResult> Handle(SapTask dequeuedTask);
-
-        protected async Task StartSession()
+        public async Task<SapLoginCookies> StartSession()
         {
-            if (SapCookies == null || DateTime.Now > _sessionStartedAt.AddMinutes(30))
+            if (_sapCookies == null)
```

Comment:
If you remove the time validation won't you be using old cookies after 30 minutes, thus making the request fail?